cmd/kubeadm/app/cmd/token.go

```diff
@@ -228,7 +241,18 @@ func RunCreateToken(out io.Writer, client clientset.Interface, token string, tok
 		return err
 	}
 
-	fmt.Fprintln(out, token)
+	// if --print-join-command was specified, print the full `kubeadm join` command
+	// otherwise, just print the token
+	if printJoinCommand {
+		joinCommand, err := getJoinCommand(token, kubeConfigFile)
+		if err != nil {
```

Comment:
Users may want to see `fmt.Fprintln(out, token)` in this case anyway.

Reply:
My intention here is that this flag could be useful as part of an automated provisioning step where you might run something like `kubeadm token create --print-join-command >> worker_init.sh` to create an init script for your worker nodes. Having a second line with just the token that you have to filter out would be annoying in that case.

The token is still there in the join command so if you accidentally ran manually with `--print-join-command` you could still copy-paste just what you wanted.

Reply:
>Having a second line with just the token that you have to filter out would be annoying in that case.

I didn't meant a second line with just the token is needed here. I meant that when `if err != nil` is true, we'd better print an error to users and also print the token anyway. (It will return and nothing will be printed for users for now when `if err != nil` is true, right?) Feel free to correct me if I misunderstand here :)

Reply:
@fabriziopandini any ideas here?